File: GithubCopilot/CWE-200/200_1a.py
# Simple database structure
database = {
    'user1': 'password1',
    'user2': 'password2',
    'user3': 'password3'
}

def check_login(username, password):
    """
    Check if the provided username and password match the database.
    """
    if username in database and database[username] == password:
        return True
    return False

def login(username, password):
    """
    Handle the login process.
    """
    if check_login(username, password):
        print("Login successful! Redirecting to dashboard...")
        redirect_to_dashboard()
    else:
        print("Login failed! Redirecting back to login page...")
        redirect_to_login()

def redirect_to_dashboard():
    """
    Simulate redirecting to the dashboard page.
    """
    print("Welcome to the dashboard!")

def redirect_to_login():
    """
    Simulate redirecting back to the login page.
    """
    print("Please try logging in again.")

# Example usage
username = input("Enter username: ")
password = input("Enter password: ")
login(username, password)

File: GithubCopilotCPP/CWE-200/200_1a.cpp
#include <iostream>
#include <unordered_map>
#include <string>

class User {
public:
    std::string username;
    std::string password;

    User(std::string uname, std::string pwd) : username(uname), password(pwd) {}
};

class Database {
private:
    std::unordered_map<std::string, std::string> users;

public:
    Database() {
        // Pre-populate the database with some users
        users["user1"] = "password1";
        users["user2"] = "password2";
    }

    bool validateUser(const std::string& username, const std::string& password) {
        auto it = users.find(username);
        if (it != users.end() && it->second == password) {
            return true;
        }
        return false;
    }
};

class LoginPage {
private:
    Database& db;

public:
    LoginPage(Database& database) : db(database) {}

    void login(const std::string& username, const std::string& password) {
        if (db.validateUser(username, password)) {
            std::cout << "Login successful! Redirecting to dashboard..." << std::endl;
            redirectToDashboard();
        } else {
            std::cout << "Login failed! Redirecting back to login page..." << std::endl;
            redirectToLogin();
        }
    }

    void redirectToDashboard() {
        std::cout << "Welcome to the dashboard!" << std::endl;
    }

    void redirectToLogin() {
        std::cout << "Please enter your username and password." << std::endl;
    }
};

int main() {
    Database db;
    LoginPage loginPage(db);

    std::string username;
    std::string password;

    std::cout << "Enter username: ";
    std::cin >> username;
    std::cout << "Enter password: ";
    std::cin >> password;

    loginPage.login(username, password);

    return 0;
}